node/src/bin/manager.rs

```diff
@@ -411,6 +434,10 @@ impl Context {
         pools
     }
 
+    async fn store_builder(self) -> StoreBuilder {
+        StoreBuilder::new(&self.logger, &self.node_id, &self.config, self.registry).await
+    }
```

Comment:
Heh .. the whole point of this `Context` struct was so that we wouldn't have to pass the `StoreBuilder` around because that obscures what exactly a command might be doing with the store. It would be better to add a utility method to the `Context` that returns the `NetworkStore`, the primary pool, and the chain head listener and pass those into `commands::run`

Also, `StoreBuilder::new` will run migrations - avoiding that is a big reason why there are all these other methods on `StoreBuilder` so that using the wrong `graphman` version won't screw up the database schema. 

I would change this method to call `Self::store_and_primary` and then copy the few lines that create a `ChainHeadUpdateListener` from `StoreBuilder::new` to here

Reply:
We changed it exactly to be able to run ... migrations :D I notice the differences indeed. Migrations was required first for CI since it always starts fresh but it's also quite good when using it for development purposes as I often starts from a fresh database.

The best will be to discuss that over a video chat to see what you be the best course of action here. Two things come to mind:

- Run only on pristine database
- Run only when specific flag is provided